File: src/main/java/kr/co/dealmungchi/hotdealbatch/domain/repository/ProviderRepository.java
package kr.co.dealmungchi.hotdealbatch.domain.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;

import kr.co.dealmungchi.hotdealbatch.domain.entity.Provider;
import kr.co.dealmungchi.hotdealbatch.domain.entity.ProviderType;

public interface ProviderRepository extends JpaRepository<Provider, Long> {

    Optional<Provider> findByProviderType(ProviderType providerType);
}

File: src/main/kotlin/kr/co/dealmungchi/hotdealbatch/domain/repository/ProviderRepository.kt
package kr.co.dealmungchi.hotdealbatch.domain.repository

import kr.co.dealmungchi.hotdealbatch.domain.entity.Provider
import kr.co.dealmungchi.hotdealbatch.domain.entity.ProviderType
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

/**
 * Repository for Provider entities.
 */
interface ProviderRepository : JpaRepository<Provider, Long> {
    /**
     * Find a provider by its type.
     *
     * @param providerType The type of the provider
     * @return An Optional containing the provider if found
     */
    fun findByProviderType(providerType: ProviderType): Optional<Provider>
}